Tests/K1Tests/TestCases/Keys/PublicKey/PublicKeyGroupOperationsTests.swift

```diff
@@ -0,0 +1,157 @@
+		// y: 483ADA77 26A3C465 5DA4FBFC 0E1108A8 FD17B448 A6855419 9C47D08F FB10D4B8
+		
+		let generatorPoint: [UInt8] = [
+			// X coordinate (32 bytes)
+			0x79, 0xBE, 0x66, 0x7E, 0xF9, 0xDC, 0xBB, 0xAC,
+			0x55, 0xA0, 0x62, 0x95, 0xCE, 0x87, 0x0B, 0x07,
+			0x02, 0x9B, 0xFC, 0xDB, 0x2D, 0xCE, 0x28, 0xD9,
+			0x59, 0xF2, 0x81, 0x5B, 0x16, 0xF8, 0x17, 0x98,
+			
+			// Y coordinate (32 bytes)
+			0x48, 0x3A, 0xDA, 0x77, 0x26, 0xA3, 0xC4, 0x65,
+			0x5D, 0xA4, 0xFB, 0xFC, 0x0E, 0x11, 0x08, 0xA8,
+			0xFD, 0x17, 0xB4, 0x48, 0xA6, 0x85, 0x54, 0x19,
+			0x9C, 0x47, 0xD0, 0x8F, 0xFB, 0x10, 0xD4, 0xB8
+		]
+		
+		let expectedGenerator = try K1.Schnorr.PublicKey(rawRepresentation: generatorPoint)
+		let actualGeneratorRaw = try FFI.PublicKey.serialize(FFI.PublicKey.Wrapped.g, format: .uncompressed)
+		let actualGenerator = try K1.Schnorr.PublicKey(rawRepresentation: actualGeneratorRaw.dropFirst()) // Drop the 0x04 prefix
+		
+		XCTAssertEqual(expectedGenerator.rawRepresentation, actualGenerator.rawRepresentation)
+	}
+	
+	func testScalarMultiples() throws {
+		// Test that scalar multiples are consistent
+		let gx2 = FFI.PublicKey.Wrapped.gx2
+		let gx3 = FFI.PublicKey.Wrapped.gx3
+		let gx4 = FFI.PublicKey.Wrapped.gx4
+		let gx5 = FFI.PublicKey.Wrapped.gx5
+		let gx6 = FFI.PublicKey.Wrapped.gx6
+
+		// These are all constructed via scalar multiplication
+		let priv2 = try FFI.PrivateKey.Wrapped(scalar: 2)
+		let priv3 = try FFI.PrivateKey.Wrapped(scalar: 3)
+		let priv4 = try FFI.PrivateKey.Wrapped(scalar: 4)
+		let priv5 = try FFI.PrivateKey.Wrapped(scalar: 5)
+		let priv6 = try FFI.PrivateKey.Wrapped(scalar: 6)
+
+		XCTAssertTrue(try gx2.compare(to: priv2.publicKey))
+		XCTAssertTrue(try gx3.compare(to: priv3.publicKey))
+		XCTAssertTrue(try gx4.compare(to: priv4.publicKey))
+		XCTAssertTrue(try gx5.compare(to: priv5.publicKey))
+		XCTAssertTrue(try gx6.compare(to: priv6.publicKey))
+	}
+
+	func testBasicAddition() throws {
+		// Test that basic addition works
+		let a = FFI.PublicKey.Wrapped.gx2
+		let b = FFI.PublicKey.Wrapped.gx3
+		
+		// Test that we can add two points
+		let sum = try a + b
+		let sumCompressed = try FFI.PublicKey.serialize(sum, format: .compressed)
+		
+		// Verify the result is a valid point (33 bytes for compressed)
+		XCTAssertEqual(sumCompressed.count, 33)
+		
+		// Verify the result is different from both inputs
+		XCTAssertFalse(try sum.compare(to: a))
+		XCTAssertFalse(try sum.compare(to: b))
+	}
+
+	func testNegation() throws {
+		// Test that negation works correctly
+		let a = FFI.PublicKey.Wrapped.gx2
+		let negA = try a.negate()
+		let negNegA = try negA.negate()
+		
+		// Test that -(-a) = a
+		XCTAssertTrue(try negNegA.compare(to: a))
+		
+		// Test that a != -a (unless a is the point at infinity, which gx2 is not)
+		XCTAssertFalse(try a.compare(to: negA))
+	}
+
+	func testAdditionIsNotScalarMultiplication() throws {
+		// On secp256k1, g + g != gx2 (2*G)
```

Comment:
This test is entirely inaccurate, `// On secp256k1, g + g != gx2 (2*G)` that claim is incorrect, it is the exact opposite. `XG + XG == 2XG` for any `X`.